backend/inspirehep/tools/views.py

```diff
@@ -0,0 +1,58 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2020 CERN.
+#
+# inspirehep is free software; you can redistribute it and/or modify it under
+# the terms of the MIT License; see LICENSE file for more details.
+
+from io import BytesIO, TextIOWrapper
+
+from flask import Blueprint, abort, current_app, jsonify, request
+
+from inspirehep.files.api import current_s3_instance
+from inspirehep.records.utils import hash_data
+from inspirehep.tools.utils import get_filename, get_mimetype, get_references
+
+blueprint = Blueprint("inspirehep_tools", __name__, url_prefix="/tools")
+
+ALLOWED_EXTENSIONS = {"tex"}
+
+
+def allowed_file(filename):
+    return "." in filename and filename.rsplit(".", 1)[1].lower() in ALLOWED_EXTENSIONS
```

Comment:
so we validate on filename? I thought we wanted to select also a MIME-type optionally in the browser (which doesn't necessarily agree with the file extension).

Reply:
As @drjova said allowing just certain extensions is the common thing to do. If we want to filter by MIME-type we can do it in the frontend. We could allow some formats such as as .tex or .txt but I don't think it makes sense to allow a .js or something else. What do you think?

Reply:
See previous comment, splitting on the first dot is not a good idea.